src/library/escape.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * escape.c - Source file for escaping capability
+ * Copyright (c) 2021 Red Hat Inc.
+ * All Rights Reserved.
+ *
+ * This software may be freely redistributed and/or modified under the
+ * terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2, or (at your option) any
+ * later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING. If not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor
+ * Boston, MA 02110-1335, USA.
+ *
+ * Authors:
+ *   Radovan Sroka <rsroka@redhat.com>
+ */
+
+#include "escape.h"
+
+#include <stdio.h>
+#include <ctype.h>
+#include <string.h>
+
+#include "message.h"
+
+static int isHEX(int x)
```

Comment:
can be remade into a macro:
#define IS_HEX(X) (isxdigit(X) > 0 && !(islower(X) > 0))